### PR TITLE
FF8: Direct mode

### PR DESCRIPTION
**Pull request by myst6re**

### CODE REVIEW

File: src/ff8_opengl.cpp [hunk 1]
```diff
@@ -558,7 +555,15 @@ void ff8_init_hooks(struct game_obj *_game_object)
 	replace_function(common_externals.destroy_tex_header, ff8_destroy_tex_header);
 	replace_function(common_externals.load_tex_file, ff8_load_tex_file);
 
-	ff8_open_file = (struct ff8_file * (*)(struct ff8_file_context *, char *)) common_externals.open_file;
+	replace_function(common_externals.open_file, ff8_open_file);
+	replace_call(uint32_t(ff8_externals.fs_archive_search_filename) + 0x10, fs_archive_search_filename2);
+	//replace_call(0x51AF52, ff8_fs_archive_create_temp_archive_malloc);
+	replace_call(0x51B5F2, fs_archive_search_filename_sub_archive_sprintf);
```
Comment (myst6re) [inline src/ff8_opengl.cpp:561]: TODO: field's temp.fs archive

File: src/ff8_opengl.cpp [hunk 2]
```diff
@@ -558,7 +555,14 @@ void ff8_init_hooks(struct game_obj *_game_object)
 	replace_function(common_externals.destroy_tex_header, ff8_destroy_tex_header);
 	replace_function(common_externals.load_tex_file, ff8_load_tex_file);
 
-	ff8_open_file = (struct ff8_file * (*)(struct ff8_file_context *, char *)) common_externals.open_file;
+	replace_function(common_externals.open_file, ff8_open_file);
+	replace_call(uint32_t(ff8_externals.fs_archive_search_filename) + 0x10, ff8_fs_archive_search_filename2);
+	replace_call(ff8_externals.moriya_filesytem_open + 0x776, ff8_fs_archive_search_filename_sub_archive);
```
Comment (myst6re) [inline src/ff8_opengl.cpp:560]: TODO

File: src/ff8_opengl.cpp [hunk 3]
```diff
@@ -558,7 +555,14 @@ void ff8_init_hooks(struct game_obj *_game_object)
...
+	replace_call(ff8_externals.moriya_filesytem_open + 0x83C, ff8_fs_archive_search_filename_sub_archive2);
+	replace_function(ff8_externals._open, ff8_open);
+	replace_function(ff8_externals.fopen, ff8_fopen);
+	replace_call(ff8_externals.moriya_filesytem_free + 0x1F, ff8_fs_archive_close_file_sub_archive);
```
Comment (myst6re) [inline src/ff8_opengl.cpp:564]: TODO

**Comment by myst6re** [comment]
Ready for review

**Comment by myst6re** [comment]
Ready

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you!

**Pull request merged**